File: Networking/Data/NetworkString.cs
namespace Networking.Data
{
    public struct NetworkString
    {
        public readonly bool isEmpty;
        public readonly bool isNull;
        public readonly bool isNullOrEmpty;

        public readonly string value;

        public NetworkString(bool isEmpty, bool isNull, string value)
        {
            this.isEmpty = isEmpty;
            this.isNull = isNull;
            this.isNullOrEmpty = isEmpty || isNull;
            this.value = value;
        }

        public string GetValue(string defaultValue = "")
        {
            if (isNullOrEmpty)
                return defaultValue;

            return value;
        }
    }
}

File: Networking/Data/NetworkString.vb
Namespace Networking.Data
    Public Structure NetworkString
        Public ReadOnly isEmpty As Boolean
        Public ReadOnly isNull As Boolean
        Public ReadOnly isNullOrEmpty As Boolean

        Public ReadOnly value As String

        Public Sub New(isEmpty As Boolean, isNull As Boolean, value As String)
            Me.isEmpty = isEmpty
            Me.isNull = isNull
            isNullOrEmpty = isEmpty OrElse isNull
            Me.value = value
        End Sub

        Public Function GetValue(Optional defaultValue As String = "") As String
            If isNullOrEmpty Then Return defaultValue

            Return value
        End Function
    End Structure
End Namespace
